modules/ding_place2book/ding_place2book.install

```diff
@@ -178,10 +178,79 @@ function ding_place2book_update_7005(&$sandbox) {
   variable_del('ding_place2book_category_mappings');
 }
 
-
 /**
  * Enable migration module for existing sites.
  */
 function ding_place2book_update_7006() {
   module_enable(array('ding_place2book_migrate'));
 }
+
+/**
+ * Remove ticket links from existing passive events.
+ */
+function ding_place2book_update_7007(&$sandbox) {
+  $nodes = [];
+
+  // Get the ding_event nodes we need to consider. We only consider those with
+  // an event end-date that hasn't passed. We can't make any assumptions about
+  // whether the nodes are publsihed or not, so no condition on $node->status.
+  $query = new EntityFieldQuery();
+  $query->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'ding_event')
+    ->fieldCondition('field_ding_event_date', 'value2', gmdate('Y-m-d H:i:s'), '>');
+
+  $result = $query->execute();
+  if (isset($result['node'])) {
+    $nodes = entity_load('node', array_keys($result['node']));
+  }
+
+  if (!isset($sandbox['processed'])) {
+    $sandbox['processed'] = 0;
+    $sandbox['total'] = count($nodes);
+  }
+
+  // Potentially we have to make request to p2b for all fetched ding_event nodes
+  // so keep the amount of nodes we process each run low.
+  $limit = 10;
+  $chunk = array_slice($nodes, $sandbox['processed'], $limit);
+
+  foreach($chunk as $nid => $node) {
+    // Avoid making request if this node isn't synchronized.
+    if (!$field_place2book = ding_place2book_is_ding_event_node_synchronized($node)) {
```

Comment:
This seems to be based the existance of the `field_place2book` field. Could we check for that in the initial query? If not the we could filter away these nodes before the actual processing.

To the memory footprint low for each batch process we should remove ids which are not relevant for this step before loading the entities.

Reply:
Hmm.. that may be a good idea. I'll take a look at it 👍

Reply:
Fixed!